base/applications/rapps/loaddlg.cpp

```diff
@@ -80,6 +81,40 @@ LoadStatusString(DownloadStatus StatusParam)
     return szString;
 }
 
+#define PATH_CHAR_CLASS_LETTER      0x00000001
+#define PATH_CHAR_CLASS_DOT         0x00000004
+#define PATH_CHAR_CLASS_SEMICOLON   0x00000020
+#define PATH_CHAR_CLASS_COMMA       0x00000040
+#define PATH_CHAR_CLASS_SPACE       0x00000080
+#define PATH_CHAR_CLASS_OTHER_VALID 0x00000100
+
+#define FILENAME_VALID_CHAR ( \
+    PATH_CHAR_CLASS_LETTER      | \
+    PATH_CHAR_CLASS_DOT         | \
+    PATH_CHAR_CLASS_SEMICOLON   | \
+    PATH_CHAR_CLASS_COMMA       | \
+    PATH_CHAR_CLASS_SPACE       | \
+    PATH_CHAR_CLASS_OTHER_VALID)
+
+VOID
+UrlUnescapeAndFileNameCensorship(CStringW& str)
+{
+    WCHAR szPath[MAX_PATH];
+    DWORD cchPath = _countof(szPath);
+    UrlUnescapeW(const_cast<LPWSTR>((LPCWSTR)str), szPath, &cchPath, 0);
+
+    for (auto& ch : szPath)
+    {
+        if (ch == 0)
+            break;
+
+        if (!PathIsValidCharW(ch, FILENAME_VALID_CHAR))
+            ch = L'_';
+    }
+
+    str = szPath;
```

Comment:
What's happening there? Is `str` now getting a full copy of the contents of the `szPath` buffer?

Reply:
Yes. `str` will get the copy of `szPath`.

Reply:
ok thanks for the clarification